### PR TITLE
Reorganize instance contracts

### PR DESCRIPTION
**Pull request by rapidddenis**
- moves kv store write operations into InstanceStore
- moves premium is calculations into PricingService
- moves instance initial authorizations configuration into InstanceAuthorizationsLibs

### CODE REVIEW

File: contracts/instance/service/PricingService.sol [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+        premium.fullPremiumAmount += t;
+
+        t = (netPremiumAmount * distInfo.distributionFee.fractionalFee).toInt();
+        premium.distributionFeeVarAmount = t;
+        premium.fullPremiumAmount += t;
+
+        return premium;
+    }
+
+    function _calculateFeeAmount(
```
Comment (doerfli) [inline contracts/instance/service/PricingService.sol:237]: We should rename this to make it clearer that it does (`calculateFeeAmount` was clear in the scope of the `DistributionService`)

```suggestion
    function _calculateDistributionFees(
```

File: scripts/libs/libraries.ts [hunk 2]
```diff
@@ -177,6 +178,16 @@ export async function deployLibraries(owner: Signer): Promise<LibraryAddresses>
                 Key32Lib: key32LibAddress,
             }
         });
+
```
Comment (doerfli) [inline scripts/libs/libraries.ts:181]: when you add a new library here, please ensure that its address is also printed in the `printAddresses` function in the `deploy_all.ts` script.

File: contracts/instance/service/PricingService.sol (hunk 1 above)
Comment (rapidddenis) [reply contracts/instance/service/PricingService.sol]: done

File: scripts/libs/libraries.ts (hunk 2 above)
Comment (rapidddenis) [reply scripts/libs/libraries.ts:181]: done

**Comment by doerfli** [comment]
i assume the diagrams 
- https://github.com/etherisc/gif-next/blob/develop/docs_manual/application.md
- https://github.com/etherisc/gif-next/blob/develop/docs_manual/distribution.md 
need to be updated. If its not done within the PR, please create an issue so its not forgotten.

**Comment by rapidddenis** [comment]
closing because of error, see #317

**Pull request closed without merging**